lib/src/chart/base/axis_chart/axis_chart_data.dart

```diff
@@ -154,19 +157,28 @@ class SideTitles with EquatableMixin {
   /// It allows you to do anything you want, For example you can show icons
   /// instead of texts, because it accepts a [Widget]
   ///
-  /// [reservedSize] determines the maximum space that your titles need,
+  /// [reservedWidth] determines the maximum width that your titles need,
+  /// (All titles will stretch using this value)
+  ///
+  /// [reservedHeight] determines the maximum height that your titles need,
   /// (All titles will stretch using this value)
   ///
   /// Texts are showing with provided [interval]. If you don't provide anything,
   /// we try to find a suitable value to set as [interval] under the hood.
   SideTitles({
     bool? showTitles,
     GetTitleWidgetFunction? getTitlesWidget,
-    double? reservedSize,
+    double? reservedWidth,
+    double? reservedHeight,
     double? interval,
   })  : showTitles = showTitles ?? false,
         getTitlesWidget = getTitlesWidget ?? defaultGetTitle,
-        reservedSize = reservedSize ?? 22,
+        reservedWidth = (reservedWidth != null || reservedHeight != null)
```

Comment:
Why checking for both? 🤔 Is it necessary?
```suggestion
        reservedWidth = (reservedWidth != null)
```

Reply:
It is necessary. 
If one is set, the other is allowed to be null.  
If both are null, it falls back to the current behavior using interval and 22 for the other dimension (previously called reservedSize).